core/templates/domain/admin/beam-job-run.model.ts

```diff
@@ -0,0 +1,180 @@
+  // state may be set by the Cloud Dataflow service, as a transition from
+  // RUNNING. It may also be set via a Cloud Dataflow jobs.update call, if the
+  // job has not yet reached a terminal state.
+  'DONE' |
+  // The job has has failed. This is a terminal job state. This state may only
+  // be set by the Cloud Dataflow service, and only as a transition from
+  // RUNNING.
+  'FAILED' |
+  // The job's run state isn't specified.
+  'UNKNOWN'
+);
+
+class MatIcon {
+  constructor(
+      public readonly tooltip: string,
+      public readonly code: string,
+      public readonly color: string | null) {}
+}
+
+export interface BeamJobRunBackendDict {
+  'job_id': string;
+  'job_name': string;
+  'job_state': BeamJobRunState;
+  'job_arguments': string[];
+  'job_started_on_msecs': number;
+  'job_updated_on_msecs': number;
+  'job_is_synchronous': boolean;
+}
+
+export class BeamJobRun {
+  private matIcon: MatIcon;
+
+  constructor(
+      public readonly jobId: string,
+      public readonly jobName: string,
+      public readonly jobState: BeamJobRunState,
+      public readonly jobArguments: readonly string[],
+      public readonly jobStartedOnMsecs: number,
+      public readonly jobUpdatedOnMsecs: number,
+      public readonly jobIsSynchronous: boolean) {
+    switch (jobState) {
+      case 'RUNNING':
+        this.matIcon = new MatIcon('Running...', 'run_circle', 'accent');
+        break;
+      case 'PENDING':
+        this.matIcon = new MatIcon('Pending...', 'pending', 'accent');
+        break;
+      case 'STOPPED':
+        this.matIcon = new MatIcon('Paused', 'pause_circle', 'accent');
+        break;
+      case 'CANCELLING':
+        this.matIcon = new MatIcon('Cancelling...', 'pending', null);
+        break;
+      case 'CANCELLED':
+        this.matIcon = new MatIcon('Cancelled', 'cancel', null);
+        break;
+      case 'DRAINING':
+        this.matIcon = new MatIcon('Draining...', 'hourglass_top', 'accent');
+        break;
+      case 'DRAINED':
+        this.matIcon = new MatIcon('Drained', 'hourglass_bottom', null);
+        break;
+      case 'UPDATED':
+        this.matIcon = (
+          new MatIcon('Updated', 'published_with_changes', 'primary'));
+        break;
+      case 'DONE':
+        this.matIcon = new MatIcon('Done!', 'check_circle', 'primary');
+        break;
+      case 'FAILED':
+        this.matIcon = new MatIcon('Failed!', 'error', 'warn');
+        break;
+      case 'UNKNOWN':
+        this.matIcon = new MatIcon('Unknown', 'help', 'warn');
+        break;
+    }
+  }
+
+  getJobStatusTooltipString(): string {
+    return this.matIcon.tooltip;
+  }
+
+  getJobStatusMaterialIconCode(): string {
+    return this.matIcon.code;
+  }
+
+  getJobStatusMaterialThemeColor(): string {
+    return this.matIcon.color;
+  }
+
+  inTerminalState(): boolean {
+    switch (this.jobState) {
+      case 'CANCELLED':
+      case 'DRAINED':
+      case 'UPDATED':
+      case 'DONE':
+      case 'FAILED':
```

Comment:
Should this be a list?

Reply:
These shouldn't ever change, so I think it's OK.

Reply:
But wouldn't the code be a bit cleaner `final_states = […]` and then checking if the `final_states` contains something.

Reply:
I think the benefit here is compile-time safety again, if a value from the enum is missing here then the code will fail to compile